plugins/modules/win_psrepository_copy.ps1

```diff
@@ -0,0 +1,234 @@
+        $Property ,
+
+        [Parameter()]
+        [String[]]
+        $Include ,
+
+        [Parameter()]
+        [String[]]
+        $Exclude
+    )
+
+    Process {
+        $o = if ($Property) {
+            $InputObject.($Property)
+        }
+        else {
+            $InputObject
+        }
+
+        foreach ($inc in $Include) {
+            $imatch = $o -like $inc
+            if ($imatch) {
+                break
+            }
+        }
+
+        if (-not $imatch) {
+            return
+        }
+
+        foreach ($exc in $Exclude) {
+            if ($o -like $exc) {
+                return
+            }
+        }
+
+        $InputObject
+    }
+}
+
+function Get-ProfileDirectory {
+    <#
+        .SYNOPSIS
+        Returns DirectoryInfo objects for each profile on the system, as reported by the registry
+
+        .DESCRIPTION
+        The special "Default" profile, used as a template for newly created users, is explicitly
+        added to the list of possible profiles returned. Public is explicitly excluded.
+        Paths reported by the registry that don't exist on the filesystem are silently skipped.
+    #>
+    [CmdletBinding()]
+    [OutputType([System.IO.DirectoryInfo])]
+    param(
+        [Parameter()]
+        [String[]]
+        $Include ,
+
+        [Parameter()]
+        [String[]]
+        $Exclude
+    )
+
+    $regPL = 'HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion\ProfileList'
+
+    # note: this is a key named "Default", not the (Default) key
+    $default = Get-ItemProperty -LiteralPath $regPL | Select-Object -ExpandProperty Default
+
+    # "ProfileImagePath" is always the local side of the profile, even if roaming profiles are used
+    # This is what we want, because PSRepositories are stored in AppData/Local and don't roam
+    $profiles = (
+        @($default) +
+        (Get-ChildItem -LiteralPath $regPL | Get-ItemProperty | Select-Object -ExpandProperty ProfileImagePath)
+    ) -as [System.IO.DirectoryInfo[]]
+
+    $profiles |
+        Where-Object -Property Exists -EQ $true |
+        Select-Wildcard -Property Name -Include $Include -Exclude $Exclude
+}
+
+function Compare-Hashtable {
+    <#
+        .SYNOPSIS
+        Attempts to naively compare two hashtables by serializing them and string comparing the serialized versions
+    #>
+    [CmdletBinding()]
+    [OutputType([bool])]
+    param(
+        [Parameter(Mandatory)]
+        [hashtable]
+        $ReferenceObject ,
+
+        [Parameter(Mandatory)]
+        [hashtable]
+        $DifferenceObject ,
+
+        [Parameter()]
+        [int]
+        $Depth
+    )
+
+    if ($PSBoundParameters.ContainsKey('Depth')) {
+        $sRef = [System.Management.Automation.PSSerializer]::Serialize($ReferenceObject, $Depth)
+        $sDif = [System.Management.Automation.PSSerializer]::Serialize($DifferenceObject, $Depth)
+    }
+    else {
+        $sRef = [System.Management.Automation.PSSerializer]::Serialize($ReferenceObject)
```

Comment:
Thank you for showing me this, I always just used Export-Clixml read the file then deleted it which is not ideal.

Reply:
Ah yeah, super useful. Occasionally combine it with `Set-Clipboard` / `scb` (and then `gcb` with the `Deserialze()`) to move large complex objects around in debugging, like in and out of RDP sessions, without dealing with a file.